aea/helpers/async_utils.py

```diff
@@ -63,37 +64,54 @@ def ensure_list(value: Any) -> List:
 class AsyncState:
     """Awaitable state."""
 
-    def __init__(self, initial_state: Any = None):
+    def __init__(
+        self, initial_state: Any = None, states_enum: Optional[Container[Any]] = None
```

Comment:
state should be an enum or class imo. not Any

Reply:
I agree. In other modules, I used to annotate arguments like this as:
```
arg: Union[SomeEnum, str]
```
and then, in the body of the function:
```
SomeEnum(arg)
```
which works both for `SomeEnum` and `str`, but in the latter case, it validates whether the string is a legal value.

Reply:
it's a general-purpose class, the state can be any value, enum is optional and it can be not an enum but any container with any values inside.
this is why Any annotation is used. i see no reason to limit it by types, it does not know anything about what kind of states are set.